src/helper/service.cpp

```diff
@@ -80,9 +80,7 @@ void service::run(std::future<bool> &&exit_signal)
         // the polling interval has actually elapsed.
         auto now = std::chrono::steady_clock::now();
         if ((now - before) >= poll_interval_) {
-            if (!rc_client_->poll()) {
-                return;
-            }
+            rc_client_->poll();
```

Comment:
So I found this while checking system tests. When a payload comes as it is not valid, `poll` returns false but it does not mean we have to stop polling. The result of polling is more for tests

Reply:
Should we have a debug message instead or is that already logged somewhere else inside `poll`?

Reply:
I'll do it now

Reply:
Done